api/users.go

```diff
@@ -127,3 +127,44 @@ func (api *API) ChangePasswordHandler(ctx context.Context, w http.ResponseWriter
 
 	return models.NewSuccessResponse(jsonResponse, http.StatusAccepted, headers), nil
 }
+
+//PasswordResetHandler creates a new user and returns a http handler interface
+func (api *API) PasswordResetHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	defer req.Body.Close()
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, handleBodyReadError(ctx, err)
+	}
+
+	passwordResetParams := models.PasswordReset{}
+	err = json.Unmarshal(body, &passwordResetParams)
+	if err != nil {
+		return nil, handleBodyUnmarshalError(ctx, err)
+	}
+
+	validationErr := passwordResetParams.Validate(ctx)
+
+	if validationErr != nil {
+		return nil, models.NewErrorResponse([]error{validationErr}, http.StatusBadRequest, nil)
+	}
+
+	forgotPasswordRequest := passwordResetParams.BuildCognitoRequest(api.ClientSecret, api.ClientId)
+
+	result, err := api.CognitoClient.ForgotPassword(forgotPasswordRequest)
+	if err != nil {
+		responseErr := models.NewCognitoError(ctx, err, "ForgotPassword request from password reset endpoint")
+		if responseErr.Code == models.InternalError {
```

Comment:
cognito response has 2 error code 400 and 500, but we are checking for just one here